### ISSUE TITLE
Directions for use

### ISSUE THREAD

**Issue by Abh15h3k** [50an6xy06r6n/keyboard_reversible.pretty#1]
Hi.

I'm confused about how to use this footprint.
is there any project currently using this that i can reference?

**Comment by arcasoy** [comment]
Also would love some help understanding this

**Comment by 50an6xy06r6n** [comment]
You should just be able to assign a Pro Micro component to the footprint and route the pads as usual. When soldering your board, you jump the jumpers on the side of the PCB that faces the same way as the bottom of the Pro Micro PCB.

**Comment by orand** [comment]
I'm new to this, so please forgive the very newbie follow-up. Does that mean route to the center through-hole, or somewhere else?  And to jump the jumpers, where do I put the blob of solder: across the two adjacent center through-holes, or somewhere else?

**Comment by orand** [comment]
Never mind, I just found the answer via this video, starting at 3:00.
https://youtu.be/vr8LkjsRqZs?t=180

It wasn't clear to me from the KiCad screenshots, but there's actually a little envelope-shaped pad next to each outer through hole, and jumping the jumpers means putting a blob of solder on that to "seal the envelope". There's also an example KiCad/Ergogen project for this board at https://github.com/benvallack/onekey

And he appears to route to the center through-holes.

**Comment by glebsexy** [comment]
So just to clarify even further, the jumpers need to be soldered on both halves of the keyboard, but on different sides of the PCB, right?

Also I'd love a footrpint like this for RP2040 Zero and Xiao. Let me know if anyone has one!

**Comment by orand** [comment]
Edit: see below.  This is actually incorrect.  I was using a different version of the footprint which is reversed.

> So just to clarify even further, the jumpers need to be soldered on both halves of the keyboard, but on different sides of the PCB, right?

 Correct. It’s designed for the jumpers to be soldered on the side of the PCB that the microcontroller will be mounted on, assuming you’re mounting it face up.

**Comment by 50an6xy06r6n** [comment]
Err it's actually designed for the jumpers to be soldered on the opposite side of the PCB from the controller, assuming the "face" is the side with all the components. It'll work either way, but I'm pretty sure that's how I mapped the pins...

**Comment by orand** [comment]
Hmm, well I'm typing this on a split keyboard using the footprint shown in the video I linked above, and I soldered the jumpers on the top side of the board and then mounted hot swap socketed nice!nano v2's on top of that, with the component side face up so that I can see the components, label, and LEDs.

**Comment by 50an6xy06r6n** [comment]
Hmm which board is it? I think there might be a version floating around where the pin mappings were flipped

**Comment by orand** [comment]
I'm using this version of the footprint.
https://github.com/benvallack/ergogen/blob/master/src/footprints/promicro_pretty.js

**Comment by 50an6xy06r6n** [comment]
Yeah that one's reversed. If you're using this in KiCAD the footprint should map the component pins in the way that I described. If you're using ergogen then you just have to be careful that you use the proper pins, but it's easier to swap them if you want the other behavior. At some point I'll have time to come back and update the library

**Comment by orand** [comment]
Whoa, good to know!  I edited my comment above to help avoid confusion.
